MRI_geom/SPD_fMRI.py

```diff
@@ -0,0 +1,182 @@
+import geomstats.spd_matrices_space as spd_space
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+from sklearn.metrics import accuracy_score, confusion_matrix
+from sklearn.metrics import f1_score, precision_score, recall_score 
+from sklearn.model_selection import KFold
+from sklearn.svm import SVC
+import time
+
+DIM_SPACE = 28
+RAND_SEED = 2018
+SPACE = spd_space.SPDMatricesSpace(dimension=DIM_SPACE)
+CORR_THRESH = 0.1
+GAMMA = 1.0
+NB_GRAPHS = 86
+SIGMAS = list(np.arange(0.5, 20, 0.5))
+
+
+def import_data():
+    graphs = pd.DataFrame.from_csv("data/train_fnc.csv")
+    map_FCN = pd.DataFrame.from_csv("add_info/comp_ind_fmri.csv",
+                                    index_col=None)
+    map_FCN = map_FCN['fMRI_comp_ind'].to_dict()
+    map_FCN_r = {v: k for k, v
+                 in map_FCN.iteritems()}
+    mapping = pd.DataFrame.from_csv("add_info/" +
+                                    "rs_fmri_fnc_mapping.csv")
+    graph_labels = pd.DataFrame.from_csv("data/train_labels.csv")
+    all_graphs = [None] * n_graphs
+    all_targets = [None] * n_graphs
+    for i in range(NB_GRAPHS):
+        u = np.zeros((DIM_SPACE, DIM_SPACE))
+        for ii in range(mapping.shape[0]):
+            a, b = (mapping.iloc[ii]["mapA"], mapping.iloc[ii]["mapB"])
+            u[map_FCN_r[a], map_FCN_r[b]] = graphs.iloc[i][ii]
+            u = np.multiply(u, (u > CORR_THRESH))
+        all_graphs[i] = np.abs(u + u.T)
+        all_targets[i] = graph_labels.loc[graphs.index[i], "Class"]
+
+    np.random.seed(RAND_SEED)
+    index_train = range(NB_GRAPHS)
+    np.random.shuffle(index_train)
+    stop = int(0.85 * NB_GRAPHS)
+    labels = np.array(all_targets[index_train])
+    return graphs.iloc[:, index_train], [all_graphs[t] for t in index_train],\
+        labels, stop, index_train
+
+
+def laplacian(a):
+    d = np.diag(np.array(a.sum(1)).flatten())
+    return d-a
+
+
+def fit_kernel_cv(log_euclidean_distance, labels,
+                  sigma=None, verbose=False):
+    kf = KFold(n_splits=10)
+    perf = {}
+    conf = {}
+    k = 0
+    if sigma is not None:
+        distance = np.exp(- np.square(log_euclidean_distance)/(sigma**2))
+    else:
+        distance = log_euclidean_distance
+    for train_index, test_index in kf.split(range(NB_GRAPHS)):
+        train_index = np.array(train_index)
+        test_index = np.array(test_index)
+        x = np.array(distance[train_index, :])[:, train_index]
+        x_test = np.array(distance[test_index, :])[:, train_index]
+        clf = SVC(kernel='precomputed')
+        clf.fit(x, labels[train_index])
+        y_train = clf.predict(x)
+        if verbose:
+            print("Training accuracy:",
+                  accuracy_score(labels[train_index], y_train))
+        y_test = clf.predict(x_test)
+        perf[k] = {'acc': accuracy_score(labels[test_index], y_test),
```

Comment:
instead of keeping k, you could also just append. Here you're indexing a dictionnary with an integer - whereas it seems what you really want is a list?